flycheck.el

```diff
@@ -7684,7 +7684,25 @@ See URL `https://github.com/jimhester/lintr'."
           line-end))
   :modes ess-mode
   ;; Don't check ESS files which do not contain R
-  :predicate (lambda () (equal ess-language "S")))
+  :predicate (lambda () (and (equal ess-language "S")
+                        (flycheck-R-has-lintr (flycheck-checker-executable 'r-lintr))))
+  :verify (lambda (checker)
+            (let ((has-lintr (flycheck-R-has-lintr
+                              (flycheck-checker-executable checker))))
+              (list
+               (flycheck-verification-result-new
+                :label "lintr library"
+                :message (if has-lintr "present" "missing")
+                :face (if has-lintr 'success '(bold error)))))))
+
+(defun flycheck-R-has-lintr (R)
```

Comment:
Please use a lowercase `r`.  I know that the language name is `R`, but for the sake of consistency I'd like to stick to Emacs Lisp' naming conventions.